### ISSUE TITLE
DataGrid: Binding the header of a DataGridTemplateColumn to a window's VM property does not work

### ISSUE THREAD

**Issue by ElMostafaIdrassi** [AvaloniaUI/Avalonia#3821]
I'm trying to bind `DataGridTextColumn` `Header` to a property of the VM but to no success. At first I thought it was because I was binding `DataGrid` `Items` to an `ObservableCollection` of the VM, but even after getting rid of that, the issue persists.

Instead of showing the correct header, it shows up as `Avalonia.Data.Binding`.

![image](https://user-images.githubusercontent.com/25257376/80234980-8d067780-8648-11ea-9e78-6a639df39c61.png)

After getting rid of `DataGrid` `Items` binding and every `DataGridTextColumn` binding.

![image](https://user-images.githubusercontent.com/25257376/80236255-6b0df480-864a-11ea-8799-fa3ad1885138.png)

I tried using 3 techniques of binding to the parent `Window` VM :

- `Binding Path=DataContext.GradesHeader, ElementName=root`
- `Binding Path=DataContext.NamesHeader, RelativeSource={RelativeSource AncestorType={x:Type Window}}`
- `Binding $parent[Window].DataContext.PresenceHeader`

Am I doing something wrong ?

Repro : [Link](https://github.com/ElMostafaIdrassi/Misc/tree/master/AvaloniaApplicationTest/AvaloniaApplicationTest)

**Comment by JamRemco** [comment]
Hello world,

Ive tried to clone an build your project but it is complaining about a json file

So I'm  navigating in  the dark now, but have you tried?
```c#
<DataGridTextColumn Header="{Binding NamesHeader}"
				Binding="{Binding Name, Mode=OneWay}"
				IsReadOnly="True"
				Width="5*" />
```
I'm hust guessing, it might work

Remco

**Comment by Gillibald** [comment]
Header is no AvaloniaProperty so I don't understand how this can work

**Comment by JamRemco** [comment]
@Gillibald  Oke!,  but a warning or a error would be nice

@ElMostafaIdrassi  what Gillibald  is saying that the Header property in not bindable.
Normaly i would look it up but the API section of the avaloniaui website is offline

regards,

Remco

**Comment by ElMostafaIdrassi** [comment]
@JamRemco Thanks for the suggestion, but that was the first thing I tried that didn't work, that's why I went with the 3 other methods. 

@Gillibald I think that's it. I didn't know the `Header` was not an `AvaloniaProperty`.

Just fyi, I was able to solve the issue by using `DataGridTextColumn.Header` with a `TextBlock` which I binded to the VM property. Here is a snippet of code that worked : 

```
<DataGridTextColumn Binding="{Binding City, Mode=OneWay}" IsReadOnly="True" Width="1*" >
	<DataGridTextColumn.Header>
		<TextBlock Text="{Binding CitiesHeader}"/>
	</DataGridTextColumn.Header>
</DataGridTextColumn>
```

![image](https://user-images.githubusercontent.com/25257376/80266297-9fa1a080-868a-11ea-9282-73d9286b2841.png)


Full code is in the repro.
Closing this as it is not really a bug.
